lib/tasks/db.rake

```diff
@@ -9,13 +9,57 @@ namespace :db do
     Dir[PRODUCTION_BACKUP_TARS].sort.reverse.first
   end
 
+  def filter_toc(path)
+    # We filter the database dump, modifying its table of contents to remove
+    # elements that we do not wish to restore:
+    #
+    # - The PostGIS extension and the spatial_ref_sys table that it creates.
+    #   This extension exists in the production dump because it is always enabled
+    #   for GOV.UK Platform as a Service Postgres instances
+    #   (https://docs.cloud.service.gov.uk/deploying_services/postgresql/#add-or-remove-extensions-for-a-postgresql-service-instance).
+    #   However, we do not make use of it in our app, and if we kept it in the
+    #   dump then we would need to use a custom Postgres Docker image with the PostGIS
+    #   extension installed. So it's easier to just get rid of it.
+    #
+    # - The reassign_owned function and event trigger. If we do not get rid of
+    #   these, then something happens that causes migrations that alter columns
+    #   to fail in development, with an error like 'role
+    #   "rdsbroker_e8e8d7d2_5890_4f9d_8c9c_da43a9918e4c_manager" does not
+    #   exist'.
+    new_toc_lines = File.read(path).each_line.reject do |line|
+      line =~ /EXTENSION.*postgis|spatial_ref_sys|reassign_owned/
```

Comment:
I trust that this operation works however I'm not familiar enough with the content with a dump file to be confident. Perhaps we could include an example snippet from the actual file before and after within the commit description?

Reply:
Sure - either in the commit message or in this comment I'll put examples of the lines that we're trying to strip.

Reply:
I've updated the commit message to give some examples of what we want to keep and what we want to scrap; does it make more sense now?

Reply:
Yes that's very helpful, thank you 👍 

After thinking about this a little more, what I found myself seeking was more confidence that this code does what we think it should with a lack of familiarity of the specifics. Given more time an idea would be to extract this logic from the rake task itself and move it into a testable object that uses a fixture with an example dump. That would serve to self document the behaviour through the tests and helped future changes to this logic to be made in future.

Let's not worry about this for now given we are drawing to an end of the current sprint, it is better to have this fixed than not stuck pending!

Reply:
It would certainly be nice to be able to validate that it works. I don't know where I'd even begin if I wanted to craft a database dump that was sufficiently close to the real dump to exhibit the problems that I'm trying to address, whilst being sufficiently anonymised and reduced in size to be able to check it into the repository!